lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @watching = []
+      @exclude = []
+      @files = {}
+      set_ignore_older(opts[:ignore_older])
+      set_close_older(opts[:close_older])
+      @wf_vars = [opts[:delimiter], @ignore_older, @close_older]
+      exclude(opts[:exclude])
+    end
+
+    def logger=(loggr)
+      @logger = loggr
+      @converter.logger = loggr
+    end
+
+    def add_converter(converter)
```

Comment:
I failed in this case. I will add the (returning) self line to make it fluent
I have schizophrenia about methods with a fluent interface vs setters that are not like `attr_writer` style.
I try to use `add_*` for the fluent methods to try to communicate their chain-ability.
```ruby
Foo.new(*args)
  .add_bar(Bar.new(*bars)) # I can guess what this does
  .add_baz(Baz.new(*bazs)) # I can guess what this does
```
vs
```ruby
foo = Foo.new(*args)
foo.set_bar(Bar.new(*bars)) # I can guess what this does
# or
foo.bar(Bar.new(*bars)) # I am unsure what this does
# or
foo.baz = Baz.new(*bazs) # I can guess what this does
foo
```
I failed in this case. I will add the (returning) self line.
There is no ruby convention to follow for naming of fluent methods or for dependency injection